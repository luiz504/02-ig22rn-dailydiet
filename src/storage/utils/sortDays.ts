export const sortDays = (array: string[]) => {
  return array.sort((a, b) => {
    return new Date(b).getTime() - new Date(a).getTime()
  })
}
