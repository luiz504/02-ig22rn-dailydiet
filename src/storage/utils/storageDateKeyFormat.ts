import { format } from 'date-fns'

export function storageDateKeyFormat(date: Date): string {
  return format(date, 'yyyy/MM/dd')
}
