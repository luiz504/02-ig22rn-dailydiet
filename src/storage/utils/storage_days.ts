import AsyncStorage from '@react-native-async-storage/async-storage'
import { DAY_COLLECTION } from '../commons'

export async function getStoredDays() {
  return AsyncStorage.getItem(DAY_COLLECTION).then((storedData) =>
    storedData ? (JSON.parse(storedData) as string[]) : null,
  )
}

export async function setStoredDays(dayCollection: string[]) {
  return AsyncStorage.setItem(DAY_COLLECTION, JSON.stringify(dayCollection))
}
