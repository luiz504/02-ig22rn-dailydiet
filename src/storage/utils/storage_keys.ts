import { format, parse } from 'date-fns'

const FORMAT = 'yyyy/MM/dd'
export function storageDateKeyFormat(date: Date): string {
  return format(date, FORMAT)
}

export function storageKeyToDateParse(key: string) {
  return parse(key, FORMAT, new Date())
}
