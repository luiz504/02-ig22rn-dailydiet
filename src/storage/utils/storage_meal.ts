import AsyncStorage from '@react-native-async-storage/async-storage'
import { MEAL_COLLECTION } from '../commons'
import { Meal } from '~/models/Meal'

export type MealGroupByDay = {
  day: string
  meals: Meal[]
}

export async function getStoredMealsByDay(key: string) {
  return AsyncStorage.getItem(`${MEAL_COLLECTION}-${key}`).then((storedData) =>
    storedData ? (JSON.parse(storedData) as MealGroupByDay) : null,
  )
}

export async function setStoredMeals(key: string, meals: Meal[]) {
  const sortedMealsByHour = meals.sort(
    (a, b) => new Date(b.date).getTime() - new Date(a.date).getTime(),
  )
  const mealsCollection: MealGroupByDay = { day: key, meals: sortedMealsByHour }

  return AsyncStorage.setItem(
    `${MEAL_COLLECTION}-${key}`,
    JSON.stringify(mealsCollection),
  )
}

export async function getStoredMealsByDays(keys: string[]) {
  const keysCombined = keys.map((k) => `${MEAL_COLLECTION}-${k}`)
  const entries = await AsyncStorage.multiGet(keysCombined)
  return entries
    .map(([, value]) => (value ? JSON.parse(value) : null))
    .filter((value) => value) as MealGroupByDay[]
}

export async function deleteStoredMealCollection(key: string) {
  return AsyncStorage.removeItem(`${MEAL_COLLECTION}-${key}`)
}
