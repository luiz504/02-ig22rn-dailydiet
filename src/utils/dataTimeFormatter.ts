import * as Localization from 'expo-localization'

const userConfig = {
  uses24hoursClock:
    Localization?.getCalendars?.()?.[0]?.uses24hourClock ?? true,
  languageTag: Localization?.getLocales?.()?.[0]?.languageTag ?? 'en-US',
}

function formatDate(date: Date) {
  return new Intl.DateTimeFormat(userConfig.languageTag, {
    year: 'numeric',
    month: '2-digit',
    day: '2-digit',
  }).format(date)
}

function formatTime(time: Date) {
  return new Intl.DateTimeFormat(userConfig.languageTag, {
    hour: '2-digit',
    minute: '2-digit',
    hour12: !userConfig.uses24hoursClock,
  }).format(time)
}

function formatDateAndTime(date: Date | string) {
  const formattedFullDate = new Date(date).toLocaleDateString(
    userConfig.languageTag,
    {
      year: 'numeric',
      month: '2-digit',
      day: '2-digit',
      hour: 'numeric',
      minute: 'numeric',
    },
  )

  if (userConfig.languageTag.startsWith('en')) {
    return formattedFullDate.replace(',', ' at')
  }

  if (userConfig.languageTag.startsWith('pt')) {
    return formattedFullDate.replace(',', ' Ã s')
  }

  return formattedFullDate
}
export { userConfig, formatDate, formatTime, formatDateAndTime }
