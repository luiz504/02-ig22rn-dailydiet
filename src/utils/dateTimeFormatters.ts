import * as Localization from 'expo-localization'

const getFirstUserConfig = () => ({
  uses24hoursClock: Localization.getCalendars()[0].uses24hourClock,
  languageTag: Localization.getLocales()[0].languageTag,
})

function formatDate(date: Date) {
  const { languageTag } = getFirstUserConfig()

  return new Intl.DateTimeFormat(languageTag, {
    year: 'numeric',
    month: '2-digit',
    day: '2-digit',
  }).format(date)
}

function formatTime(time: Date) {
  const { languageTag, uses24hoursClock } = getFirstUserConfig()

  return new Intl.DateTimeFormat(languageTag, {
    hour: '2-digit',
    minute: '2-digit',
    hour12: !uses24hoursClock,
  }).format(time)
}

function formatDateAndTime(date: Date | string) {
  const { languageTag, uses24hoursClock } = getFirstUserConfig()

  const formattedFullDate = new Date(date).toLocaleDateString(languageTag, {
    year: 'numeric',
    month: '2-digit',
    day: '2-digit',
    hour: '2-digit',
    minute: '2-digit',
    hour12: !uses24hoursClock,
  })

  if (languageTag.startsWith('en')) {
    return formattedFullDate.replace(',', ' at')
  }

  if (languageTag.startsWith('pt')) {
    return formattedFullDate.replace(',', ' Ã s')
  }

  return formattedFullDate
}

export { getFirstUserConfig, formatDate, formatTime, formatDateAndTime }
