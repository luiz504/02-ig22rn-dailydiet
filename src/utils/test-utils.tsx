import { FC, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components/native'
import { render } from '@testing-library/react-native'

import { theme } from '~/styles'
import { NavigationContainer } from '@react-navigation/native'

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  )
}

const ThemedWrapper: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender: typeof render = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const renderThemed: typeof render = (ui, options) =>
  render(ui, { wrapper: ThemedWrapper, ...options })

const splitBorderColor = (borderColor: string) => ({
  borderTopColor: borderColor,
  borderRightColor: borderColor,
  borderBottomColor: borderColor,
  borderLeftColor: borderColor,
})

export * from '@testing-library/react-native'

export { customRender as render, renderThemed, splitBorderColor }
