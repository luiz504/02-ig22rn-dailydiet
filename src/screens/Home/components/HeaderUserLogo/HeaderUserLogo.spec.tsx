import { render, screen } from '~/utils/test-utils'

import { HeaderUserLogo } from '.'

describe('HeaderUserLogo Component', () => {
  it('should render correctly', () => {
    render(<HeaderUserLogo />)

    expect(screen.getByTestId('header-user-logo').props.style).toMatchSnapshot(
      'container-styles',
    )

    expect(screen.getByTestId('logo')).toBeOnTheScreen()
    expect(screen.getByTestId('profile').props.style).toMatchSnapshot(
      'profile-styles',
    )
  })
})
