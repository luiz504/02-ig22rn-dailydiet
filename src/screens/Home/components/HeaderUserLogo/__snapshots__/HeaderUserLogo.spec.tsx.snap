// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderUserLogo Component should render correctly: container-styles 1`] = `
{
  "alignItems": "center",
  "flexDirection": "row",
  "justifyContent": "space-between",
  "paddingBottom": 24,
  "paddingLeft": 0,
  "paddingRight": 0,
  "paddingTop": 24,
}
`;

exports[`HeaderUserLogo Component should render correctly: profile-styles 1`] = `
{
  "borderBottomLeftRadius": 20,
  "borderBottomRightRadius": 20,
  "borderColor": "#333638",
  "borderStyle": "solid",
  "borderTopLeftRadius": 20,
  "borderTopRightRadius": 20,
  "borderWidth": 2,
  "height": 40,
  "width": 40,
}
`;
